server/src/apub/extensions/signatures.rs

```diff
@@ -35,28 +35,29 @@ pub fn generate_actor_keypair() -> Result<Keypair, Error> {
   })
 }
 
+// TODO is it possible to create this signature, with just the url and actor?
 /// Signs request headers with the given keypair.
-pub fn sign(request: &Builder, actor: &dyn ActorType) -> Result<String, Error> {
+pub fn sign(request: &mut RequestBuilder, actor: &dyn ActorType) -> Result<String, Error> {
   let signing_key_id = format!("{}#main-key", actor.actor_id());
 
   let headers = request
-    .headers_ref()
-    .unwrap()
+    .inspect()
+    .headers()
     .iter()
     .map(|h| -> Result<(String, String), Error> {
       Ok((h.0.as_str().to_owned(), h.1.to_str()?.to_owned()))
     })
     .collect::<Result<BTreeMap<String, String>, Error>>()?;
 
+  let mut path_and_query = request.inspect().url().path().to_owned();
+  if let Some(query) = request.inspect().url().query() {
+    path_and_query.push_str(query);
+  }
+
```

Comment:
It could've been this path_and_query I had to manually build, since that doesn't exist on `Url`

Reply:
Okay looking at the `signing_string` used for signing and verification, the problem is because signing doesnt include the port in `host`, but verification does include it. So either you need to include the port here, or exclude it below in the verification. Not sure which option is better, anyway there should never be a different port in production. Maybe a helper method to handle this, which includes the port only in debug mode? Or just ignore the port completely.

Reply:
Hrm... the `to_url.domain()` should probably just pass in the url and port right. Easier to just include the port, so it works for all cases.